examples/substrate-react-example/src/App.tsx

```diff
@@ -0,0 +1,68 @@
+import {
+  JSXElementConstructor,
+  ReactElement,
+  ReactFragment,
+} from "react";
+import "./App.css";
+
+import { SubstrateContextProvider, useSubstrateState } from "./substrate-lib";
+
+import UserInfo from "./components/UserInfo";
+import Form from "./components/Form";
+import TransferStatus from "./components/TransferStatus"
+
+const Loader = (
+  text:
+    | string
+    | number
+    | boolean
+    | ReactElement<unknown, string | JSXElementConstructor<unknown>>
+    | ReactFragment
+    | null
+    | undefined
+) => <div>{text}</div>;
+
+const Message = (errObj: { target: { url: string } }) => (
+  <div>{`Connection to websocket '${errObj.target.url}' failed.`}</div>
+);
+
+function Main() {
+  const { apiState, apiError, keyringState, keyring } = useSubstrateState()!;
+
+  if (apiState === "ERROR") return Message(apiError as { target: { url: string } });
+  if (apiState !== "READY") return Loader("Connecting to Substrate");
+
+  if (keyringState === "ERROR" && !keyring) {
+    return Loader(
+      "Can't get access to etensions accounts. Please authorize in extension"
+    );
+  }
+
+  if (keyringState !== "READY") {
+    return Loader(
+      "Loading accounts (please review any extension's authorization)"
+    );
+  }
```

Comment:
The code block with error handling could be improved by using constants for the status. It could also be separated into another method for error handling. 

What is the possible error state when using "!== READY"? Can we compare the exact values as it gives the feeling of some potential unknown keyring value?